lib/puppet/property.rb

```diff
@@ -346,6 +362,21 @@ def insync?(is)
     end
   end
 
+  # This method tests if two values are insync? outside of the properties current
+  # should value. This works around the requirement for corrective_change analysis
+  # that requires two older values to be compared with the properties potentially
+  # custom insync? code.
+  #
+  # @param [Object] orig_should the original should value
+  # @param [Object] orig_is the original is value
+  # @return [Boolean] whether or not the values are in sync or not
+  # @api private
+  def insync_values?(orig_should, orig_is)
+    test_property = self.dup
```

Comment:
I'm also not super thrilled with this, but I agree with your reasoning, Ken :(

If we were dup'ing a property in a more complex situation I'd be a lot more scared, but given that this is tiny function and the lifetime of the dup'd property is easy to understand, I think it's OK-ish. Just don't let it escape into the wild where somebody might call #sync on it, or we're gonna have a situation

![](https://media.giphy.com/media/eLpoGELHst13a/giphy.gif)